src/lib.rs

```diff
@@ -98,19 +100,20 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            Error::Decode(ref e) => fmt::Display::fmt(e, f),
+            Error::Decode(ref e) => write!(f, "{e}"),
```

Comment:
In 72a190f36c143626572f4679e258cc1d34dbd6fb:

You are changing this to override any format specifications on `f` (e.g. alternate display). This is a mild reduction in functionality (e.g. it prevents pretty-printing these variants) and also isn't mentioned in the commit message. What is the purpose of this?

Reply:
I didn't know about this, so thank you for pointing this out. I thought `write!(f, "{e}")` is more readable and is expanded into `fmt::Display::fmt(e, f)`.

Reply:
If you want, you can replace `fmt::Display::fmt(e, f)` with `e.fmt(f)` and Rust knows to call the right `fmt`. (I don't know how it knows this. It's quite mysterious to me.)